src/python/pants/backend/docker/lint/hadolint/rules_integration_test.py

```diff
@@ -0,0 +1,150 @@
+    FROM python:3.8
+    """
+)
+
+BAD_FILE = dedent(
+    """
+    FROM python
+    """
+)
+
+
+def run_hadolint(
+    rule_runner: RuleRunner, targets: list[Target], *, extra_args: list[str] | None = None
+) -> tuple[LintResult, ...]:
+    rule_runner.set_options(
+        ["--backend-packages=pants.backend.docker.lint.hadolint", *(extra_args or ())],
+        env_inherit={"PATH"},
+    )
+    results = rule_runner.request(
+        LintResults,
+        [HadolintRequest(HadolintFieldSet.create(tgt) for tgt in targets)],
+    )
+    return results.results
+
+
+def assert_success(
+    rule_runner: RuleRunner, target: Target, *, extra_args: list[str] | None = None
+) -> None:
+    result = run_hadolint(rule_runner, [target], extra_args=extra_args)
+    assert len(result) == 1
+    assert result[0].exit_code == 0
+    assert not result[0].stdout
+    assert not result[0].stderr
+
+
+def test_passing(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files({"Dockerfile": GOOD_FILE, "BUILD": "docker_image(name='t')"})
+    tgt = rule_runner.get_target(Address("", target_name="t"))
+    assert_success(rule_runner, tgt)
+
+
+def test_failing(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files({"Dockerfile": BAD_FILE, "BUILD": "docker_image(name='t')"})
+    tgt = rule_runner.get_target(Address("", target_name="t"))
+    result = run_hadolint(rule_runner, [tgt])
+    assert len(result) == 1
+    assert result[0].exit_code == 1
+    assert "Dockerfile:2 " in result[0].stdout
+
+
+def test_multiple_targets(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "Dockerfile.good": GOOD_FILE,
+            "Dockerfile.bad": BAD_FILE,
+            "BUILD": dedent(
+                """
+                docker_image(name="good", sources=("Dockerfile.good",))
+                docker_image(name="bad", sources=("Dockerfile.bad",))
+                """
+            ),
+        }
+    )
+    tgts = [
+        rule_runner.get_target(
+            Address("", target_name="good", relative_file_path="Dockerfile.good")
+        ),
+        rule_runner.get_target(Address("", target_name="bad", relative_file_path="Dockerfile.bad")),
+    ]
+    result = run_hadolint(rule_runner, tgts)
+    assert len(result) == 1
+    assert result[0].exit_code == 1
+    assert "Dockerfile.good" not in result[0].stdout
+    assert "Dockerfile.bad:2 " in result[0].stdout
+
+
+def test_config_files(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "a/Dockerfile": BAD_FILE,
+            "a/BUILD": "docker_image()",
+            "a/.hadolint.yaml": "ignored: [DL3006]",
+            "b/Dockerfile": BAD_FILE,
+            "b/BUILD": "docker_image()",
+        }
+    )
+    tgts = [
+        rule_runner.get_target(Address("a")),
+        rule_runner.get_target(Address("b")),
+    ]
+    result = run_hadolint(rule_runner, tgts)
+    # We get two runs of hadolint, for the `a` and `b` directories respectively.
```

Comment:
Due to there being a config file in `a`, otherwise it would have been a single run.